src/integrations/prefect-dask/prefect_dask/client.py

```diff
@@ -54,6 +54,14 @@ def submit(
                 }
             run_task_kwargs["dependencies"] = dependencies
 
+            asset_context = AssetContext.from_task_and_inputs(
+                task=func, task_run_id=task_run_id, task_inputs=dependencies
+            )
+            asset_context.update_tracked_assets()
+            context = serialize_context()
+            context["asset_context"] = asset_context.serialize()
```

Comment:
i think `serialize_context` exists to consolidate this logic so not every runner has to re-implement it; I highly recommend putting the `asset_context.serialize()` call within `serialize_context`.

Reply:
In this case I very specifically do not want to pass the current `AssetContext` and instead want to ship off a different `AssetContext`.

Would you prefer if I update `serialize_context` to accept overrides?

Reply:
gotcha -- if that behavior is what you think is necessary for distributed / remote environments in general, then I think you can hard-code new context creation within the utility.